### ISSUE TITLE
Screen reader is not announcing a new tab opens on APIs overview links

### ISSUE THREAD

**Issue by claudiahub** [backstage/backstage#11102]
## Expected Behavior

Screen reader notifies the user the needed information when page opens in new tab for 'View Source' link, 'Edit' button and 'Slack Channel' link.

## Actual Behavior

1. Navigate to APIs from the side menu bar.
2. Navigate to the main section of the page (table component).
3. Select any link from the name column. A new page opens up.
4. Launch a screen reader and navigate to 'View Source' link, 'Edit' button and 'Slack Channel' link.
5. Verify whether the screen reader on focus announces that the clicked UI control opens in new tab or not.

## Context

Screen reader users aren't aware of clickable links and some buttons that opens in a new tab as the ones mentioned above.

A11y Conformance Level: A

WCAG Success Criteria: 3.2.2

## Environment

Browser Information: 
Google Chrome v100.0.4896.127
Firefox 99.0.1
Safari 15.4

Screen readers used:
NVDA v2021.3

**Comment by claudiahub** [comment]
This is happening for any external link in Backstage.

WCAG reference [https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G201](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G201)
